### ISSUE TITLE
Error object 'num_compounds.this_species' not found when generating custom database

### ISSUE THREAD

**Issue by jamesrco** [vanmooylipidomics/LOBSTAHS#6]
Posted for @hfredricks. User is encountering error
```
Error in FUN(newX[, i], ...) : 
  object 'num_compounds.this_species' not found
```
when attempting to generate a new database from custom .csv tables. Similar to error encountered by earlier user (issue https://github.com/vanmooylipidomics/LOBSTAHS/issues/2) when attempting to add a lipid class not accommodated by `generateLOBdbase()`.

**Comment by jamesrco** [comment]
Error appears to be of same origin. @hfredricks is attempting to create a database that included some lipid species not yet accommodated by the `generateLOBdbase()` function. The time has  come to fix the the package so users can fully define a new compound or lipid class using only the input tables, so that modifications do not need to be made to the `generateLOBdbase()` function every time a new compound is added. This will require the addition of a few new fields to the basic component definitions table (https://github.com/vanmooylipidomics/LOBSTAHS/blob/master/inst/doc/csv/LOBSTAHS_basic_component_matrix.csv) and some modifications to the function code that make it a bit more generic.

**Comment by jamesrco** [comment]
Fixes incorporated into new package version 1.1.2. Makes adding new lipid classes and compounds to custom databases **much** easier.

Available now on GitHub: https://github.com/vanmooylipidomics/LOBSTAHS/releases/tag/v1.1.2 and pushing soon to Bioconductor-devel

Directions and explanations in revised vignette: https://github.com/vanmooylipidomics/LOBSTAHS/blob/master/vignettes/LOBSTAHS.Rmd

**Comment by hfredricks** [comment]
Thanks Jamie!
